src/debugger/android/androidPlatform.ts

```diff
@@ -7,14 +7,29 @@ import {IRunOptions} from "../launchArgs";
 import {CommandExecutor} from "../../common/commandExecutor";
 import {Package} from "../../common/node/package";
 import {PackageNameResolver} from "../../common/android/packageNameResolver";
+import {MakeOutcomeFailDependingOnOutput, PatternToFailure} from "../../common/MakeOutcomeFailDependingOnOutput";
 
 /**
  * Android specific platform implementation for debugging RN applications.
  */
 export class AndroidPlatform implements IAppPlatform {
+    // We should add the common Android build/run erros we find to this list
+    private static RUN_ANDROID_FAILURE_PATTERNS: PatternToFailure = {
+        "Failed to install on any devices": "Could not install the app on any available device. Make sure you have a correctly"
+         + " configured device or emulator running. See https://facebook.github.io/react-native/docs/android-setup.html",
+    "com.android.ddmlib.ShellCommandUnresponsiveException": "An Android shell command timed-out. Please retry the operation.",
+    "Android project not found": "Android project not found.",
+    "error: more than one device/emulator": "You have more than a single device/emulator running. Please make sure you have a single one (you can verify it by running adb devices)" };
+
+    private static RUN_ANDROID_SUCCESS_PATTERNS: string[] = ["BUILD SUCCESSFUL", "Starting the app", "Starting: Intent"];
```

Comment:
It's not unless we go and read their source code, and make sure that our list of failures is 100% complete. I don't think it's worth the effort to do that, this is more an "heuristic" rather than an exact solution, until we can convince react-native to return proper error codes in run-android and run-ios.